kernel/src/syscall.rs

```diff
@@ -114,4 +116,16 @@ pub trait UserspaceKernelBoundary {
         stack_pointer: *const usize,
         state: &mut Self::StoredState,
     ) -> (*mut usize, ContextSwitchReason);
+
+    /// Display any general information about the fault.
+    unsafe fn fault_str(&self, writer: &mut Write);
+
+    /// Display architecture specific (e.g. CPU registers or status flags) data
+    /// for a process identified by its stack pointer.
+    unsafe fn print_process_arch_detail(
```

Comment:
my only complaint is the difference in naming between this method ( `print_process_arch_detail`) and `fault_str`. They do similar things---write a formatted string out to a `Write`, but one is `print_` and the other is `_str`. Maybe `fault_fmt` and `process_detail_fmt`?

Reply:
Changed.